File: software/examples/c/example_callback.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_joystick.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change XYZ to the UID of your Joystick Bricklet

// Callback function for pressed callback
void cb_pressed(void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Pressed\n");
}

// Callback function for released callback
void cb_released(void *user_data) {
	(void)user_data; // avoid unused parameter warning

	printf("Released\n");
}

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Joystick j;
	joystick_create(&j, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Register pressed callback to function cb_pressed
	joystick_register_callback(&j,
	                           JOYSTICK_CALLBACK_PRESSED,
	                           (void (*)(void))cb_pressed,
	                           NULL);

	// Register released callback to function cb_released
	joystick_register_callback(&j,
	                           JOYSTICK_CALLBACK_RELEASED,
	                           (void (*)(void))cb_released,
	                           NULL);

	printf("Press key to exit\n");
	getchar();
	joystick_destroy(&j);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletJoystick.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletJoystick;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

// Callback function for pressed callback
function cb_pressed()
{
    echo "Pressed\n";
}

// Callback function for released callback
function cb_released()
{
    echo "Released\n";
}

$ipcon = new IPConnection(); // Create IP connection
$j = new BrickletJoystick(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register pressed callback to function cb_pressed
$j->registerCallback(BrickletJoystick::CALLBACK_PRESSED, 'cb_pressed');

// Register released callback to function cb_released
$j->registerCallback(BrickletJoystick::CALLBACK_RELEASED, 'cb_released');

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
